<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <!-- Arguments -->
  <arg name="agent_name" default="a1_gazebo"/>
  <arg name="initial_ts_state_from_agent" default="False"/>
  <arg name="obstacle_names" default="[]" />

  <!-- Load parameters -->
  <!-- Agent name -->
  <param name="agent_name" type="string" value="$(arg agent_name)" />
  <!-- hard and soft task parameters -->
  <rosparam command="load" file="$(find ltl_automation_a1)/config/a1_ltl_formula.yaml" />
  <!-- Transition system -->
  <param name="transition_system_textfile" type="str" textfile="$(find ltl_automation_a1)/config/a1_ts.yaml" />

  <!-- Run LTL planner node -->
  <include file="$(find ltl_automaton_planner)/launch/ltl_planner.launch" >
    <arg name="initial_ts_state_from_agent" default="False"/>
  </include>

  <!-- Run Nexus LTL node with the 2D pose region monitor-->
  <node name="ltl_automation_a1_node" pkg="ltl_automation_a1" type="ltl_automation_a1_node2.py" output="screen" >
        <rosparam param="obstacle_names" subst_value="True">$(arg obstacle_names)</rosparam> 
  </node>
  <node name="$(arg agent_name)_region_2d_pose_monitor" pkg="ltl_automaton_std_transition_systems" type="region_2d_pose_monitor.py" output="screen" >
  	<remap from="agent_2d_region_pose" to="qualisys/a1_gazebo/pose" />
  </node>

</launch>
